airbyte-config/persistence/src/main/java/io/airbyte/config/persistence/ConfigRepository.java

```diff
@@ -168,7 +170,14 @@ public SourceConnection getSourceConnection(final UUID sourceId) throws JsonVali
     return persistence.getConfig(ConfigSchema.SOURCE_CONNECTION, sourceId.toString(), SourceConnection.class);
   }
 
-  public void writeSourceConnection(final SourceConnection source) throws JsonValidationException, IOException {
+  public void writeSourceConnection(final SourceConnection source, final ConnectorSpecification connectorSpecification)
+      throws JsonValidationException, IOException {
+    System.out.println("connectorSpecification = " + connectorSpecification);
+    // START: ONLY FOR SANITY CHECKING I'M PASSING CORRECT VALUES
```

Comment:
i don't think it's crazy for you to keep this unless it is getting checked somewhere else. trying to keep our sanity on correctness here might be a good thing.